tests/unit/builders/indexable-builder-test.php

```diff
@@ -290,27 +385,54 @@ public function test_build_for_id_and_type_with_post_given_and_author_indexable_
 			->once()
 			->with( 1337 );
 
-		Monkey\Actions\expectDone( 'wpseo_save_indexable' )
-			->once()
-			->with( $this->indexable, $this->indexable );
-
 		$this->hierarchy_builder
 			->expects( 'build' )
 			->once()
 			->with( $this->indexable );
 
 		$author_indexable = Mockery::mock( Indexable_Mock::class );
+
+		$author_indexable
+			->expects( 'as_array' )
+			->once()
+			->andReturn( [] );
+
 		$this->indexable_repository
 			->expects( 'find_by_id_and_type' )
 			->once()
 			->with( 1999, 'user', false )
 			->andReturn( $author_indexable );
 
+		$this->indexable_repository
+			->expects( 'create' )
+			->twice()
+			->with( [] )
+			->andReturn( $this->indexable, $author_indexable );
+
+		$this->version_manager
+			->expects( 'indexable_needs_upgrade' )
+			->once()
+			->withAnyArgs()
+			->andReturnTrue();
+
+		$this->version_manager
+			->expects( 'set_latest' )
+			->twice()
+			->withAnyArgs()
+			->andReturnUsing(
+				function ( $indexable ) {
+					$indexable->version = 2;
+					return $indexable;
+				}
+			);
+
+		// Prevent the indexable builder from actually saving in this test.
+		// The test is complex enough in its current state.
 		$this->indexable_helper
 			->expects( 'should_index_indexables' )
-			->once()
+			->twice()
 			->withNoArgs()
-			->andReturnTrue();
+			->andReturnFalse();
```

Comment:
this class will be changed in the near future, so that methods will become smaller and unit tests will grow smaller too. If we'd add Partials now, we'd have to remove them later on. I'd like to leave it as it is for now, given that it won't be here long (famous last words)